packages/p2p-v2/src/components/Modals/FilterModal/FilterModalContent/FilterModalContent.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import clsx from 'clsx';
+import { Text } from '@deriv-com/ui';
+import { FilterModalPaymentMethods } from '../FilterModalPaymentMethods';
+
+type TFilterModalContentProps = {
+    filterOptions: {
+        component: JSX.Element;
+        onClick?: () => void;
+        subtext: string;
+        text: string;
+    }[];
+    paymentMethods: string[];
+    setPaymentMethods: (value: string[]) => void;
+    showPaymentMethods: boolean;
+};
+
+const FilterModalContent = ({
+    filterOptions,
+    paymentMethods,
+    setPaymentMethods,
+    showPaymentMethods,
+}: TFilterModalContentProps) => {
+    return (
+        <>
+            {showPaymentMethods ? (
+                <FilterModalPaymentMethods
+                    selectedPaymentMethods={paymentMethods}
+                    setSelectedPaymentMethods={setPaymentMethods}
+                />
+            ) : (
+                <>
+                    {filterOptions.map(option => (
+                        <div
+                            className={clsx(
+                                'py-8 px-[1.6rem] flex flex-row items-center justify-between border-b-[1px] border-solid border-[#f2f3f4]',
+                                {
+                                    'cursor-pointer': option.onClick,
+                                }
+                            )}
+                            key={option.text}
+                            {...(option.onClick && { onClick: option.onClick })}
```

Comment:
Would this work? You can still pass undefined to onClick
```suggestion
                             { onClick: option.onClick })
```

Reply:
aight let me check